### PR TITLE
Feature/react refactoring

### PR DESCRIPTION
**Pull request by ddelpiano**
VFB Component initialisation refactoring.

1. VFB wrapper and toolbar integrated in the extension itself:
 -- to integrate html viewer for the 3 menus about/contribute/feedback
 -- to update the links related to the template (to be discussed with @Robbie1977  if he prefers the nomenclature i= or id= in the url, that is the only difference).
2. ComponentInitialization routing to VFBMain component.
3. VFB Main components handle all the children needed for the UI with all the logic.
4. Stack Viewer and Term Info have been slightly reworked but they are just a wrap of what was in use before, these 2 will be re-worked with other epic cards already opened.
5. Tutorial Widget wrapping the tutorial using widget capability.

Open issue:

Query and spotlight link to finish, should finish this today.
Wireframe action in the button bar to link, done for today as well, very easy.
Resize to be tested and fixed accordingly if any issue is present.

### CODE REVIEW

**Comment by Robbie1977** [comment]
i= is the list of visual items to load and id= is the one to ensure is shown in term info after the load finishes

**Comment by Robbie1977** [comment]
See Issues: #253 & #254

File: components/VFBMain.js [hunk 1]
```diff
@@ -0,0 +1,1225 @@
...
+            }
+        }.bind(this));
+    };
+
+    // Children handlers
+    buttonBarHandler(buttonState) {
+        if((buttonState !== "infoBtn") || (buttonState !== "stackBtn")) {
+            var tempState = this.state[buttonState];
+            this.setState({
+                [buttonState] : !tempState
```
Comment (tarelli) [inline components/VFBMain.js]: I see what you are trying to do here saving some code and why the states are called what they are, the downside of this though is that the logic is less readable. For anybody who hasn't written the code they will look at those states and won't know what they do just by reading their name. 

infoBtn: true vs termInfoVisible: true
stackBtn: true vs stackViewerVisible: true
tutorialBtn: false vs tutorialVisible: false
etc.

One way you could do this is renaming the ids in the buttonBar configuration to be just "tutorial" and inside this function append the string "visible" before accessing the state. In general it's a fine line with these "smart" tricks before we write unnecessarily hard to read code so better to always be careful and err on the side of simplicity at the cost of more lines.

File: components/VFBMain.js [hunk 2]
```diff
@@ -0,0 +1,1225 @@
...
+import ButtonBar from '../../../js/components/interface/buttonBar/ButtonBar';
+import SpotLight from '../../../js/components/interface/spotlight/spotlight';
+import LinkButton from '../../../js/components/interface/linkButton/LinkButton';
+import ControlPanel from '../../../js/components/interface/controlPanel/controlpanel';
+
+
+// import NewWidget from '../../../js/components/widgets/NewWidget'; For future use.
+require('./VFBMain.less');
+require('../css/VFB.css');
+var $ = require('jquery');
```
[the remark this answers is not in the document]
Comment (ddelpiano) [reply components/VFBMain.js]: resolved bringing what needed in VFB.css into VFBMain.less to have a single file.

File: ComponentsInitialization.js [hunk 3]
```diff
@@ -1,1562 +1,25 @@
...
-                    }
-                    return compositeInstances;
-                });
-                // custom controls configuration in the controls column
-                GEPPETTO.ControlPanel.setControlsConfig({
-                    "VisualCapability": {
-                        "select": {
-                            "id": "select",
-                            "condition": "GEPPETTO.SceneController.isSelected($instance$.$instance$_obj != undefined ? [$instance$.$instance$_obj] : []) ||  GEPPETTO.SceneController.isSelected($instance$.$instance$_swc != undefined ? [$instance$.$instance$_swc] : [])",
-                            "false":
```
[the remark this answers is not in the document]
Comment (ddelpiano) [reply ComponentsInitialization.js]: not sure I understand this point, sorry.

File: components/VFBMain.js [hunk 4]
```diff
@@ -0,0 +1,1225 @@
...
+            Project.loadFromURL(window.location.origin.replace('https:','http:') + '/' + GEPPETTO_CONFIGURATION.contextPath + '/geppetto/extensions/geppetto-vfb/model/vfb.json');
+            // Local deployment for development. Uncomment the line below.
+            // Project.loadFromURL(window.location.origin.replace(":8081", "") + '/' + 'project/vfb.json');
+            this.setState({modelLoaded: true});
+        }
+    };
+
+    componentDidMount() {
+        // Global functions linked to VFBMain functions
+        window.stackViewerRequest = function(idFromStack) {
```
[the remark this answers is not in the document]
Comment (ddelpiano) [reply components/VFBMain.js]: card opened.

File: components/VFBMain.js [hunk 5]
```diff
@@ -0,0 +1,1225 @@
...
+        this.controlpanelRender = undefined;
+        this.querybuilderRender = undefined;
+        //this.logoStyle = { fontSize: '33px'};
+        this.tutorialsList = [
+            "/org.geppetto.frontend/geppetto/extensions/geppetto-vfb/tutorials/queryTutorial.json",
+            "/org.geppetto.frontend/geppetto/extensions/geppetto-vfb/tutorials/spotlightTutorial.json",
+            "/org.geppetto.frontend/geppetto/extensions/geppetto-vfb/tutorials/stackTutorial.json",
+            "/org.geppetto.frontend/geppetto/extensions/geppetto-vfb/tutorials/termTutorial.json"];
+
+        this.buttonBarConfig = {
```
[the remark this answers is not in the document]
Comment (ddelpiano) [reply components/VFBMain.js]: this was already part of my plan initially :)

File: components/interface/StackWidget.js [hunk 6]
```diff
@@ -0,0 +1,245 @@
+import React, { Component } from 'react';
+
+export default class StackWidget extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            stackViewOpened: true,
```
[the remark this answers is not in the document]
Comment (ddelpiano) [reply components/interface/StackWidget.js]: done 👍

File: components/VFBMain.js (hunk 1 above)
Comment (ddelpiano) [reply components/VFBMain.js]: done 👍

**Pull request closed without merging**